src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -138,16 +139,36 @@ public ReduceableSearchResult reduce(Collection<Collector> collectors) {
         }
 
         if (!hybridTopScoreDocCollectors.isEmpty()) {
-            HybridTopScoreDocCollector hybridTopScoreDocCollector = hybridTopScoreDocCollectors.stream()
-                .findFirst()
-                .orElseThrow(() -> new IllegalStateException("cannot collect results of hybrid search query"));
-            List<TopDocs> topDocs = hybridTopScoreDocCollector.topDocs();
-            TopDocs newTopDocs = getNewTopDocs(
-                getTotalHits(this.trackTotalHitsUpTo, topDocs, isSingleShard, hybridTopScoreDocCollector.getTotalHits()),
-                topDocs
-            );
-            TopDocsAndMaxScore topDocsAndMaxScore = new TopDocsAndMaxScore(newTopDocs, hybridTopScoreDocCollector.getMaxScore());
-            return (QuerySearchResult result) -> { result.topDocs(topDocsAndMaxScore, getSortValueFormats(sortAndFormats)); };
+            List<ReduceableSearchResult> results = new ArrayList<>();
+            for (HybridTopScoreDocCollector hybridTopScoreDocCollector : hybridTopScoreDocCollectors) {
+                List<TopDocs> topDocs = hybridTopScoreDocCollector.topDocs();
+                TopDocs newTopDocs = getNewTopDocs(
+                    getTotalHits(this.trackTotalHitsUpTo, topDocs, hybridTopScoreDocCollector.getTotalHits()),
+                    topDocs
+                );
+                TopDocsAndMaxScore topDocsAndMaxScore = new TopDocsAndMaxScore(newTopDocs, hybridTopScoreDocCollector.getMaxScore());
+
+                results.add((QuerySearchResult result) -> {
+                    // this is case of first collector, query result object doesn't have any top docs set, so we can
+                    // just set new top docs without merge
+                    if (result.hasConsumedTopDocs()) {
+                        result.topDocs(topDocsAndMaxScore, getSortValueFormats(sortAndFormats));
+                        return;
+                    }
+                    // in this case top docs are already present in result, and we need to merge next result object with what we have.
```

Comment:
I think the comment on line 158 should be present on line 162.

Reply:
nope, that comment signifies the whole section where we do have something in query result and we may need to merge it with new object